### PR TITLE
Verify trackers's parameters in trackingInitialize()

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Verify trackers's parameters in trackingInitialize().

### Checklist
- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

File: src/provider/provider_tracking.c [hunk 1]
```diff
@@ -147,10 +147,6 @@ static umf_result_t trackingAlloc(void *hProvider, size_t size,
         (umf_tracking_memory_provider_t *)hProvider;
     umf_result_t ret = UMF_RESULT_SUCCESS;
 
-    if (!p->hUpstream) {
```
Comment (bratpiorka) [inline src/provider/provider_tracking.c:150]: why do you remove that?

Comment (ldorau) [reply src/provider/provider_tracking.c:150]: Because this is checked once at the beginning in `trackingInitialize()`. We can have an assert here instead.

Comment (ldorau) [reply src/provider/provider_tracking.c:150]: assert added instead

Comment (bratpiorka) [reply src/provider/provider_tracking.c:150]: ok looks good for me

**Review by bratpiorka — APPROVED** [review]

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**